File: javascript/bob/bob.js
/* eslint-disable no-unused-vars */
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

class Bob {
  onlyNumber(str) {
    str = str.replace(/[\:\,\s]+/g, '')
    return Number.isInteger(Number(str))
  }

  hey(message) {
    message = message.trim()
    if (!message) {
      return 'Fine. Be that way!'
    }

    if (message[message.length - 1] === '?') {
      if (message === message.toUpperCase() && !this.onlyNumber(message[0])) {
        return 'Calm down, I know what I\'m doing!'
      }
      return 'Sure.'

      // if (message[message.length - 1] === '?' && )
      // if (!Number(message[0])) {

      // }
    }

    // if (message[message.length - 1] === '?') {
    //   return 'Sure.'
    // }

    if (message === message.toUpperCase() && !this.onlyNumber(message)) {
      return 'Whoa, chill out!'
    }

    return 'Whatever.'
  }
}

export default Bob;

File: ruby/bob/bob.rb
class Bob
  def self.only_n(str)
    mystr = str
    mystr.gsub!(/[\:\,\)\s]+/, '')
    return Float(mystr) != nil rescue false
  end
  def self.hey(str)
    str.strip!
    if (str.empty?)
      return "Fine. Be that way!"
    elsif (str[-1] == '?')
      if (str.upcase == str && only_n(str) == false && str.match(/[A-Z]+/))
        return "Calm down, I know what I'm doing!"
      end
      return "Sure."
    elsif (str.upcase == str && only_n(str) == false)
      return "Whoa, chill out!"
    else
      return "Whatever."
    end
  end
end